### PR TITLE
[SPIR-V] Add partial order tests, assert reducible

### PR DESCRIPTION
**Pull request by Keenuts**
Add testing for the visitor and added a note explaining irreducible CFG are not supported.
Related to #116692

### CODE REVIEW

File: llvm/test/CodeGen/SPIRV/structurizer/cf.switch.ifstmt.simple2.ll [hunk 1]
```diff
@@ -20,21 +19,21 @@
...
 ;       b += 5;
 ;       break;
 ;     case 6: {
-;     case 7:
-;       break;}
+;               case 7:
+;                 break;}
 ;     default:
-;       break;
+;             break;
```
Comment (s-perron) [inline llvm/test/CodeGen/SPIRV/structurizer/cf.switch.ifstmt.simple2.ll:36]: Was this change intended?

**Review by s-perron — APPROVED** [review]

**Review by VyacheslavLevytskyy — APPROVED** [review]

File: llvm/test/CodeGen/SPIRV/structurizer/cf.switch.ifstmt.simple2.ll (hunk 1 above)
Comment (Keenuts) [reply llvm/test/CodeGen/SPIRV/structurizer/cf.switch.ifstmt.simple2.ll]: no, error when doing a clang-format of the code. Reverted this part.

Comment (Keenuts) [reply llvm/test/CodeGen/SPIRV/structurizer/cf.switch.ifstmt.simple2.ll]: Other PR had the same change. This PR will revert those format changes.

**Comment by Keenuts** [comment]
rebased on main, will merge once green.

**Comment by damyanp** [comment]
@Keenuts - I think that the build failures above, that are timeouts, are related to this change.  I also see a SPIRVTests.exe process get stuck when I run the tests locally.

**Comment by Keenuts** [comment]
> @Keenuts - I think that the build failures above, that are timeouts, are related to this change. I also see a SPIRVTests.exe process get stuck when I run the tests locally.

Thanks!
There is a test that is expected to hit an assert (otherwise it hangs). Do you build with asserts enabled?
(I might need to replace this assert with another way to die)

**Comment by VyacheslavLevytskyy** [comment]
yes, let's please do smth like report_fatal_error instead of the assert if it hangs

**Comment by Keenuts** [comment]
Sent out https://github.com/llvm/llvm-project/pull/118617
